app/services/providers/copy_to_recruitment_cycle_service.rb

```diff
@@ -0,0 +1,67 @@
+module Providers
+  class CopyToRecruitmentCycleService
+    def initialize(provider:, copy_course_to_provider_service:, copy_site_to_provider_service:)
+      @provider = provider
+      @copy_course_to_provider_service = copy_course_to_provider_service
+      @copy_site_to_provider_service = copy_site_to_provider_service
+    end
+
+    def execute(new_recruitment_cycle)
+      providers_count = 0
+      sites_count = 0
+      courses_count = 0
+
+      ActiveRecord::Base.transaction do
+        new_provider = new_recruitment_cycle
+                          .providers
+                          .find_by(provider_code: @provider.provider_code)
+
+        unless new_provider
```

Comment:
Similar to elsewhere - this check could be made clearer. I _think_ this is ensuring we're not copying something that already exists, but the name `new_provider` has me confused.

Would be better as `unless provider_code_already_exists_on_recruitment_cycle?`